Automattic/wp-calypso
# Performance Profiler: Pass the device used to LLM query

Resolves https://github.com/Automattic/dotcom-forge/issues/10106

## Proposed Changes

* Adds the selected Tab, either `Mobile` or `Desktop` to a React context so it can be shared without _prop drilling_
* Changes the Typescript type `enum` by an object - More Info here: p7H4VZ-3sK-p2
* Unifies the usage of the `TabType` type between logged-in and logged-out versions
* Passes the device selected to the LLM query

## Why are these changes being made?


* The device type will be used by the LLM for better insights
* Both versions will use the same Type for the device selected
* Other inner components could use the device strategy, for example, for track events without requiring a lot of prop passing.

## Testing Instructions



* Apply this patch
* Navigate to the Performance Profiler and run a report, or open an existing report, e.g. `/speed-test-tool?url=https%3A%2F%2Fwordpress.com%2F&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6Nzc3Mn0.048zsfIxmJ0M_YpPGRg0QSsTmtMglyBAlPMJhrGB1vg`
* Change between Desktop and Mobile and check the results are the expected ones
* Scroll down and open an Insight
* Check the Developer Tools > Network tab for the `/odie/assistant/performance-profiler` endpoint passes a `device_strategy` field
* Repeat the steps with the Logged-in version of the tool

## Pre-merge Checklist
